pkg/ingester/ingester_test.go

```diff
@@ -608,6 +610,114 @@ func TestIngester_Push(t *testing.T) {
 				cortex_ingester_tsdb_head_active_appenders 0
 			`,
 		},
+		"should discard metadata when max metadata per user exceeded": {
+			maxMetadataPerUser:   1,
+			maxMetadataPerMetric: 0,
+			reqs: []*mimirpb.WriteRequest{
+				{
+					Metadata: []*mimirpb.MetricMetadata{
+						{Type: mimirpb.COUNTER, MetricFamilyName: "test_metric_1", Help: "This is a test metric."},
+						{Type: mimirpb.COUNTER, MetricFamilyName: "test_metric_2", Help: "This is a test metric."},
+						{Type: mimirpb.COUNTER, MetricFamilyName: "test_metric_3", Help: "This is a test metric."},
+						{Type: mimirpb.COUNTER, MetricFamilyName: "test_metric_4", Help: "This is a test metric."},
+					},
+				},
+			},
+			expectedErr:      nil,
+			expectedIngested: nil,
+			expectedMetadataIngested: []*mimirpb.MetricMetadata{
+				// we expect 3 (rather than the limit of 1) because we scale the global limit by replicationFactor/numIngesters
```

Comment:
I find this very confusing. Also we're testing with 1 ingester, when RF=3 requires at least 3. I would rather set `cfg.IngesterRing.ReplicationFactor = 1` below, right after `cfg := defaultIngesterTestConfig(t)`. This will make the test easier to understand.